src/components/Progress/Progress.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {useEffect, forwardRef, useCallback, useState, memo} from 'react'
+import {View, LayoutChangeEvent} from 'react-native'
+import {metrics, deviceWidth} from '../../helpers/metrics'
+import styled from 'styled-components/native'
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+  withRepeat,
+  Easing,
+  interpolate,
+} from 'react-native-reanimated'
+import {colors} from '../../helpers/colors'
+
+interface IProgressProps {
+  /**
+   * Current percent of the Progress bar
+   * default 0, max 100
+   */
+  value?: number
+  /**
+   * Defines height of Progress bar
+   * default 16
+   */
+  size?: number
+  /**
+   * Defines borderRadius of Progress bar
+   * default 0
+   */
+  borderRadius?: number
+  /**
+   * Defines color of Track Bar
+   */
+  filledTrackColor?: string
+  /**
+   * Defines background color of Progress bar
+   */
+  backgroundColor?: string
+  /**
+   * Defines full width of the progress bar
+   */
+  width?: number
+  /**
+   * Defines progress mode
+   */
+  isIndeterminateProgress?: boolean
+}
+
+type ProgressStyle = {
+  width: number | undefined
+  size: number | undefined
+  backgroundColor: string
+  borderRadius: number
+}
+
+const screenWidth = deviceWidth()
+const MAX_VALUE = 100
+
+const Progress = forwardRef<View, IProgressProps>(
+  (
+    {
+      width,
+      value = 0,
+      size = metrics.small,
+      borderRadius = 0,
+      filledTrackColor = colors.primary,
```

Comment:
use color from theme

Reply:
how to use theme on forwardRef props

Reply:
color is import {colors} from '../../helpers/colors' 

import {theme} from '../../theme'
I mean we change to use color from the theme, ex: theme.colors.primary